packages/neon/neon/lib/src/widgets/sync_status_icon.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:flutter/material.dart';
+import 'package:material_design_icons_flutter/material_design_icons_flutter.dart';
+import 'package:neon/src/sync/mapping.dart';
+import 'package:neon/src/theme/colors.dart';
+
+class SyncStatusIcon extends StatelessWidget {
+  const SyncStatusIcon({
+    required this.status,
+    this.size,
+    super.key,
+  });
+
+  final SyncMappingStatus status;
+  final double? size;
+
+  @override
+  Widget build(final BuildContext context) {
+    final (icon, color) = switch (status) {
+      SyncMappingStatus.unknown => (MdiIcons.cloudQuestion, NcColors.error),
+      SyncMappingStatus.incomplete => (MdiIcons.cloudSync, NcColors.warning),
+      SyncMappingStatus.complete => (MdiIcons.cloudCheck, NcColors.success),
+    };
```

Comment:
cosmetic stuff but a newline between the return and the assignment would be nice